src/components/dialog.js

```diff
@@ -261,17 +264,46 @@ export class Dialog {
     this.getContainer().appendChild(view.getElement());
 
     // If the current view should fade the parent document.
-    if (view.shouldFadeBody()) {
+    if (view.shouldFadeBody() && !this.hidden_) {
       this.graypane_.show(/* animate */ true);
     }
+
     return view.init(this).then(() => {
       setImportantStyles(view.getElement(), {
         'opacity': 1,
       });
+      if (this.hidden_ && view.shouldFadeBody()) {
+        this.graypane_.show(/* animated */ true);
+      }
+      this.show_();
+      this.hidden_ = false;
```

Comment:
`this.hidden_ = false`, imho, should be set inside the `show_()` method.

Reply:
done